.github/workflows/build_linux_packages.yml

```diff
@@ -91,9 +97,18 @@ jobs:
         run: |
           ./build_tools/fetch_sources.py --jobs 12
 
-      - name: Install python deps
+      - name: Install python and dev deps
         run: |
           pip install -r requirements.txt
+          pip install -r requirements-dev.txt
+
+      - name: Configure AWS Credentials
+        if: always()
+        uses: aws-actions/configure-aws-credentials@ececac1a45f3b08a01d2dd070d28d111c5fe6722 # v4.1.0
```

Comment:
@geomin12 we're getting a lot of these `if: always()` and `if: ${{ !cancelled() }}` conditions in these workflows. Let's see if we can restructure the flow a bit through migrating to more scripts such that the flow is easier to understand.

Here in particular, if fetching sources fails, we'd still configure AWS credentials and try to upload logs, but we wouldn't try to build. In that case I would prefer for the workflow to fail fast and not continue to try cleanup steps that will fail.